thavalon-server/src/lobby/lobby_impl.rs

```diff
@@ -0,0 +1,213 @@
+use super::client::{OutgoingMessage, PlayerClient};
+use super::{LobbyChannel, LobbyCommand, LobbyError, LobbyResponse, ResponseChannel};
+use crate::database::games::{DBGameError, DBGameStatus, DatabaseGame};
+use crate::game::builder::GameBuilder;
+use crate::utils;
+
+use tokio::{
+    sync::mpsc::{self, Receiver},
+    task,
+};
+use warp::filters::ws::WebSocket;
+
+use std::collections::HashMap;
+
+/// A lobby for an individual game. The Lobby acts as an interface between the
+/// Thavalon game instance, the DatabaseGame which keeps the game state in sync
+/// with the database, and all players connected to the game.
+pub struct Lobby {
+    database_game: DatabaseGame,
+    friend_code: String,
+    player_ids_to_client_ids: HashMap<String, String>,
+    clients: HashMap<String, PlayerClient>,
```

Comment:
Out of curiosity, why use client IDs in addition to player IDs?

Reply:
There was a reason, but I honestly am not remembering it now. Maybe it was a bad reason.

Reply:
I think you said player ids (or was it client ids?) should be private while other is public?

Reply:
Oh yeah. We use the client ID to generate a unique link to connect to the game and don't have any validation on the user after that point. If we use player ID for this, someone could steal a player's ID and impersonate them for every game. The player ID is "public" as part of the JWT. If someone got your player ID, they could just connect to any game as you. Using a random client ID stops this.